kotlinx-coroutines-core/jvm/test/ThreadContextElementTest.kt

```diff
@@ -169,15 +172,36 @@ class ThreadContextElementTest : TestBase() {
         }
     }
 
+    /**
+     * For stability of the test, it is important to make sure that
+     * the parent job actually suspends when calling
+     * `withContext(dispatcher2 + CoroutineName("dispatched"))`.
+     *
+     * Here this requirement is fulfilled by forcing execution on a single thread.
+     * However, dispatching is performed with two non-equal dispatchers to simulate multithreaded behavior.
```

Comment:
You probably mean "to avoid dispatch short-circuit" instead of "simulate multithreaded behavior.", otherwise it's a bit confusing

Reply:
Actually my goal was to keep force dispatching here: `withContext(dispatcher2 + CoroutineName("dispatched"))`. If we use only one or two equal dispatchers, then there will be no dispatch at this point